Source/Urho3D/Network/AbstractConnection.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020-2025 the rbfx project.
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT> or the accompanying LICENSE file.
+
+#include "Urho3D/Precompiled.h"
+
+#include "Urho3D/Network/AbstractConnection.h"
+
+namespace Urho3D
+{
+
+void AbstractConnection::SendMessage(
+    NetworkMessageId messageId, ConstByteSpan payload, PacketTypeFlags packetType, ea::string_view debugInfo)
+{
+    if (payload.size() > GetMaxMessageSize())
+    {
+        URHO3D_LOGERROR("{}: Message #{} ({} bytes) is too big to send", ToString(), static_cast<unsigned>(messageId),
+            payload.size());
+        return;
+    }
+
+    SendMessageInternal(messageId, payload.data(), payload.size(), packetType);
+
+    Log::GetLogger().Write(GetMessageLogLevel(messageId), "{}: Message #{} ({} bytes) sent{}{}{}{}", ToString(),
+        static_cast<unsigned>(messageId), payload.size(), (packetType & PacketType::Reliable) ? ", reliable" : "",
+        (packetType & PacketType::Ordered) ? ", ordered" : "", debugInfo.empty() ? "" : ": ", debugInfo);
```

Comment:
As i understand we will be paying formatting cost for every single packet, which seems to be a bit overkill. Currently we always log either as debug or trace log levels. That in itself is probably an overkill. At very least i would suggest adding a special case here where `LOG_NONE` skips logging code altogether. And maybe consider improving logging levels. Default to `LOG_NONE` with a way to override maybe. Idk it is up to you.